src/main/java/com/spring/demo/controllers/DemoController.java

```diff
@@ -14,30 +20,40 @@
 
 @RestController
 @RequestMapping("/api")
+@CrossOrigin
 public class DemoController {
 
+  private final ItemGateway itemGateway;
+
+  @Autowired
+  public DemoController(ItemServiceDiskImpl itemServiceDisk, ItemGateway itemGateway) {
+    this.itemGateway = itemGateway;
+  }
+
   @GetMapping(path = "/items")
   public List<Item> getItems(@RequestParam(required = false) String id) {
-    return Arrays.asList(new Item());
+    id = StringUtils.defaultString(id);
```

Comment:
don't need to make temporary variable for this value
`return itemGateway.getItems(ItemChannels.API_HEADER_VALUE_GET_ITEMS, StringUtils.defaultString(id));`